pkg/logs/launchers/file/launcher.go

```diff
@@ -117,10 +123,21 @@ func (s *Launcher) run() {
 		case source := <-s.removedSources:
 			s.removeSource(source)
 		case <-scanTicker.C:
-			s.cleanUpRotatedTailers()
-			// check if there are new files to tail, tailers to stop and tailer to restart because of file rotation
-			s.scan()
+			if s.scanMutex.TryLock() {
+				s.wg.Add(1)
+				go func() {
+					defer s.scanMutex.Unlock()
+					defer s.wg.Done()
+
+					s.cleanUpRotatedTailers()
```

Comment:
IMO this dones't need to be in the goroutine and we can do this synchronously as it's not an expensive operation.

Reply:
Moved out 👍